### PR TITLE
Feature/114.comparison operators

### LINKED ISSUE #114: Comparison operators
To make if statement usefull let's implement `>, <, >=, <=, ==, !=, &&, ||` operators — section 6.5.8,9,13,14 of standard

### PR DESCRIPTION
**Pull request by Griboedoff**
Naive implementation of comparison operators. 
I think || and && need optimization.

Closes #114.

### CODE REVIEW

File: Cesium.CodeGen/Ir/Expressions/BinaryOperatorExpression.cs [hunk 1]
```diff
@@ -25,7 +26,22 @@ public BinaryOperatorExpression(Ast.BinaryOperatorExpression expression)
...
+            BinaryOperator.EqualTo,
+            new ConstantExpression(new IntegerConstant("0"))),
+        BinaryOperator.LessThanOrEqualTo => new BinaryOperatorExpression(
+            new BinaryOperatorExpression(Left.Lower(), BinaryOperator.GreaterThan, Right.Lower()),
+            BinaryOperator.EqualTo,
+            new ConstantExpression(new IntegerConstant("0"))),
+        BinaryOperator.NotEqualTo => new BinaryOperatorExpression(
+            new BinaryOperatorExpression(Left.Lower(), BinaryOperator.EqualTo, Right.Lower()),
+            BinaryOperator.EqualTo,
+            new ConstantExpression(new IntegerConstant("0"))),
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Expressions/BinaryOperatorExpression.cs:42]: All these definitions will break for `NaN` when we support floats 😈 

For now, it's fine I guess, but let's open a new issue to not forget to handle it properly.

Comment (ForNeVeR) [reply Cesium.CodeGen/Ir/Expressions/BinaryOperatorExpression.cs:42]: Done in #117.

**Comment by Griboedoff** [comment]
I changed comment to issue link

**Comment by ForNeVeR** [comment]
Thanks!

**Review by ForNeVeR — APPROVED** [review]

**Pull request merged**